packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/network/ProgressiveStringDecoderTest.kt

```diff
@@ -0,0 +1,145 @@
+      "\uD800\uDE55\uD800\uDE55\uD800\uDE55 \uD800\uDE55\uD800\uDE55\uD800\uDE55\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80" +
+      "\uD800\uDE55\uD800\uDE55\uD800\uDE55\uD800\uDE55\uD800\uDE55\uD800\uDE55\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80" +
+      "\uD800\uDE55\uD800\uDE55\uD800\uDE55\uD800\uDE55\uD800\uDE55\uD800\uDE55\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80\uD800\uDE80" +
+      "\uD800\uDE80\uD800\uDE80\uD800\uDE80")
+
+  @Test
+  fun testUTF8SingleByteSymbols() {
+    chunkString(TEST_DATA_1_BYTE, StandardCharsets.UTF_8, 64)
+  }
+
+  @Test
+  fun testUTF8twoBytesSymbols() {
+    chunkString(TEST_DATA_2_BYTES, StandardCharsets.UTF_8, 63)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testUTF8ThreeBytesSymbols() {
+    chunkString(TEST_DATA_3_BYTES, StandardCharsets.UTF_8, 64)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testUTF8FourBytesSymbols() {
+    chunkString(TEST_DATA_4_BYTES, StandardCharsets.UTF_8, 111)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testUTF16LEStandard() {
+    chunkString(TEST_DATA_3_BYTES, StandardCharsets.UTF_16LE, 47)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testUTF16LESurrogates() {
+    // 4 bytes UTF-8 symbols are encoded as two 2 byte surrogate symbols in UTF-16
+    chunkString(TEST_DATA_4_BYTES, StandardCharsets.UTF_16LE, 47)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testUTF16BEStandard() {
+    chunkString(TEST_DATA_3_BYTES, StandardCharsets.UTF_16BE, 47)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testUTF16BESurrogates() {
+    // 4 bytes UTF-8 symbols are encoded as two 2 byte surrogate symbols in UTF-16
+    chunkString(TEST_DATA_4_BYTES, StandardCharsets.UTF_16BE, 47)
+  }
+
+  @Test
+  @Throws(Exception::class)
+  fun testUTF32() {
+    // UTF-32 data symbols always 4 bytes
+    chunkString(TEST_DATA_4_BYTES, Charset.forName("UTF-32"), 65)
+  }
+
+  private fun chunkString(originalString: String, charset: Charset, chunkSize: Int) {
+    val data = originalString.toByteArray(charset)
+    val builder = StringBuilder()
+    val collector = ProgressiveStringDecoder(charset)
+    val buffer = ByteArray(chunkSize)
+    var i = 0
+    while (i < data.size) {
+      val bytesRead = Math.min(chunkSize, data.size - i)
```

Comment:
```suggestion
      val bytesRead = min(chunkSize, data.size - i)
```

Let's use `min` from Kotlin stdlib